### DISCUSSION TITLE
[Feature] Execute command on all hosts in a group

### DISCUSSION THREAD

**Discussion by I-Want-ToBelieve** [grafviktor/goto#68] [Ideas]
just like pssh

**Comment by grafviktor** [comment]
Hi @I-Want-ToBelieve . Thank you for raising this! That would be a really cool feature to have, indeed. However, I need to think if it is possible integrate such functionality seamlessly into the existing "UI" 🙂

It's much easier to extend functionality when it's a purely command line application(because it's just an extra command line option) but when an app has even a basic UI, it's harder to add side-features without distracting users from the main functionality. In other words, features like that should not pollute the interface, but at the same time they should be easily accessible and intuitive.

For now, that's how I see it (have to warn you here, that I'm not a very smart guy, sorry) - when the app, supports hosts grouping (see #66), then we would be able to run something like that: 
```bash
# Syntax
# gg <group name> <command to execute>

# Example:
gg production_machines echo "hello, world"
```

What do you think? Please let me know if you have more\better ideas 🙂

**Reply by I-Want-ToBelieve** [reply]
I think what you said makes sense.

 Considering the design of TUI, it is a wise choice to put functions in subcommands temporarily or permanently.

**Reply by grafviktor** [reply]
OK, thank you very much for the confirmation. I will leave it in a format of discussion for some time in case if either of us (or maybe someone else) has alternative idea how it should work. I will convert this discussion back to issue right before ready for the implementation.

I must admit, that it will be actioned not early than after several months from now as it depends on groupings support (#66). And groupings is quite a big task. No false promises from my side! 🙂 )))

**Reply by I-Want-ToBelieve** [reply]
Ok, thank you.
